src/iuse.cpp

```diff
@@ -6459,20 +6559,800 @@ struct npc_photo_def : public JsonDeserializer, public JsonSerializer {
+    }
+    if( !obj_coll.terrain.empty() ) {
+        std::string obj_list = enumerate_as_string( obj_coll.terrain.begin(), obj_coll.terrain.end(),
+                               format_object_pair_article );
+        photo_text += "\n\n" + ( obj_coll.terrain.size() == 1 ?
+                                 string_format( _( "There is %s in the background." ), obj_list ) :
+                                 string_format( _( "There are %s in the background." ), obj_list ) );
+    }
+
+    const oter_id &cur_ter = overmap_buffer.ter( ms_to_omt_copy( g->m.getabs( aim_point ) ) );
+    std::string overmap_desc = string_format( _( "In the background you can see a %s" ),
+                               colorize( cur_ter->get_name(), cur_ter->get_color() ) );
+    if( outside_tiles_num == total_tiles_num ) {
+        photo_text += _( "\n\nThis photo was taken <color_gray>outside</color>." );
+    } else if( outside_tiles_num == 0 ) {
+        photo_text += _( "\n\nThis photo was taken <color_gray>inside</color>." );
+        overmap_desc += _( " interior" );
+    } else if( outside_tiles_num < total_tiles_num / 2.0 ) {
+        photo_text += _( "\n\nThis photo was taken mostly <color_gray>inside</color>,"
+                         " but <color_gray>outside</color> can be seen." );
+        overmap_desc += _( " interior" );
+    } else if( outside_tiles_num >= total_tiles_num / 2.0 ) {
+        photo_text += _( "\n\nThis photo was taken mostly <color_gray>outside</color>,"
+                         " but <color_gray>inside</color> can be seen." );
+    }
+    photo_text += "\n" + overmap_desc + ".";
+
+    if( g->get_levz() >= 0 && need_store_weather ) {
+        photo_text += "\n\n";
+        if( calendar::turn.is_sunrise_now() ) {
+            photo_text += _( "It is <color_yellow>sunrise</color>. " );
+        } else if( calendar::turn.is_sunset_now() ) {
+            photo_text += _( "It is <color_light_red>sunset</color>. " );
+        } else if( calendar::turn.is_night() ) {
+            photo_text += _( "It is <color_dark_gray>night</color>. " );
+        } else {
+            photo_text += _( "It is day. " );
+        }
+
+        const weather_datum w_data = weather_data( g->weather.weather );
+        photo_text += string_format( _( "The weather is %s." ), colorize( w_data.name, w_data.color ) );
+    }
+
+    for( const auto &figure : description_figures_appearance ) {
+        photo_text += "\n\n<color_light_blue>" + figure.first + "</color>" + _( " appearance:" ) + "\n" +
+                      figure.second;
+    }
+
+    photo_text += "\n\n" + string_format( _( "The photo was taken on %s." ),
+                                          "<color_light_blue>" + timestamp + "</color>" );
+
+    photo.description = photo_text;
+
+    return photo;
+}
+
+static void item_save_monsters( player *p, item *it, const std::vector<monster *> &monster_vec,
```

Comment:
Please change `p` and `it` to be references. They are currently dereferenced unconditionally, which means the function does simply not accept null pointers. Changing them to references makes that clear to the reader.